zoo/entities/builder.py

```diff
@@ -0,0 +1,107 @@
+from zoo.entities.models import Entity, Group, Link
+from zoo.libraries.models import Library
+from zoo.services.models import Environment, Service
+
+
+class EntityBuilder:
+    def create_entity(self, data, repository):
+        if data["kind_"] != "component":
+            return NotImplemented
+        if data["spec"]["type_"] == "service":
+            self._build_service(data, repository)
+        elif data["spec"]["type_"] == "library":
+            self._build_library(data, repository)
+        else:
+            self._build_base_component(data, repository)
```

Comment:
do we have this use case? of storing just a base component? 🤔

Reply:
I only see `create_entity` called with entities, for the base we already have `_build_base_component`

Reply:
IIRC we were talking about letting people store also other types than service and library but we wont be storing any `spec` info about them... for now i added the `database` only, but we can definitely think about more of this types. So that was my use case for base component

Reply:
I don't like this if chain here. I would personally replace it with a registry pattern and create builders for service and library as separate classes. Then pass the registry to the Entity builder to use it. This way adding new component builders will be simple to add and simple to understand.

Reply:
One personal rule I have is "Private methods usually indicate that those should be extracted as public methods of a separate class"

Reply:
it mentioned in TODO as i didnt want to spend more time on this, could be refactored in future when adding new types/kinds

Reply:
👍